v1/messages/src/requestHandlers.ts

```diff
@@ -41,66 +50,98 @@ async function handlePostRequest(message: string): Promise<HttpResponse> {
     body: JSON.stringify({
       messageId: id,
       createdAt: createdAt,
-      message: message,
+      message: requestBody.message,
     }),
   };
 }
 
 async function handleDeleteRequest(msgID: string): Promise<HttpResponse> {
   const conn = Sqlite.openDefault();
-  const id = msgID;
 
-  if (ifExists(conn, id) == false) {
+  if (!ifExists(conn, msgID)) {
     return { status: 404 };
   }
 
-  await conn.execute("DELETE FROM messages WHERE id = ?", [id]);
+  await conn.execute("DELETE FROM messages WHERE id = ?", [msgID]);
 
   return { status: 200 };
 }
 
 async function handlePutRequest(
-  newMessage: string,
+  requestBody: MessagePayload,
   msgID: string
 ): Promise<HttpResponse> {
+  if (requestBody.message === undefined) {
+    return { status: 400 };
+  }
   const conn = Sqlite.openDefault();
-  const id = msgID;
 
-  if (ifExists(conn, id) == false) {
+  if (!ifExists(conn, msgID)) {
     return { status: 404 };
   }
 
   await conn.execute("UPDATE messages SET message = ? WHERE id = ?", [
-    newMessage,
-    id,
+    requestBody.message,
+    msgID,
   ]);
 
   return { status: 200 };
 }
 
 function ifExists(conn: any, id: string): boolean {
-  const table = conn.execute("SELECT * FROM messages WHERE id = ?", [id]);
-  return table.rows.length > 0;
+  return (
+    conn.execute("SELECT * FROM messages WHERE id = ?", [id]).rows.length > 0
+  );
 }
 
-function parseSlugFromURI(uri: string, resource: string): string {
-  const uriParts = uri.split("/");
-  const resourceIndex = uriParts.indexOf(resource);
+function parseSlugFromURI(headers: Record<string, string>): string {
+  const urlParts = headers["spin-full-url"].split("/");
+  const resourceIndex = urlParts.indexOf(COMPONENT_NAME);
 
   if (resourceIndex != -1) {
-    const msgID = uriParts[resourceIndex + 1] ?? "";
+    const msgID = urlParts[resourceIndex + 1] ?? "";
     if (msgID.length > 0) {
       return msgID;
     }
   }
   return "";
 }
 
+function parseUserID(auth: string | undefined): string | null {
```

Comment:
nit: I would modify this function to take in the request header object directly, and then get the authorization header from within this function